src/Gui/Placement.cpp

```diff
@@ -559,6 +572,83 @@ void Placement::showDefaultButtons(bool ok)
     }
 }
 
+bool Placement::eventFilter(QObject*, QEvent* ev) {
+    //handle spacenav daemon motion events in placement window
+    if (//event->type() == Spaceball::ButtonEvent::ButtonEventType ||
+        ev->type() == Spaceball::MotionEvent::MotionEventType){
+                Spaceball::MotionEvent *motionEvent = dynamic_cast<Spaceball::MotionEvent*>(ev);
+                if (!motionEvent) {
+                    Base::Console().Log("invalid spaceball motion event in bool Placement::eventFilter(QObject*, QEvent* ev)\n");
+                    return false;
+                    }
+                motionEvent->setHandled(true);
+
+                static double translationConstant(.001f);
+                double xTrans = static_cast<double>(motionEvent->translationX())*translationConstant;
+                double yTrans = static_cast<double>(motionEvent->translationY())*translationConstant;
+                double zTrans = static_cast<double>(motionEvent->translationZ())*translationConstant;
+
+                static double rotationConstant(.001f);
+                double dY=static_cast<double>(motionEvent->rotationZ()) * rotationConstant;
+                double dP=static_cast<double>(motionEvent->rotationY()) * rotationConstant;
+                double dR=static_cast<double>(motionEvent->rotationX()) * rotationConstant;
+
+                if(xTrans == 0 && yTrans == 0 && zTrans == 0 && dY ==0 && dP ==0 && dR ==0 ) return false;
+                signalMapper->blockSignals(true);

```

Comment:
I recommend to use: `QSignalBlocker blocker(signalMapper);` because this is exception-safe and you don't have to worry about unblocking it when leaving the method (RAII).